master/internal/checkpoint_gc.go

```diff
@@ -49,6 +59,7 @@ func (t *checkpointGCTask) Receive(ctx *actor.Context) error {
 		for _, a := range msg.Allocations {
 			taskSpec := *t.taskSpec
 			taskSpec.AgentUserGroup = t.agentUserGroup
+			taskSpec.TaskToken = taskToken
```

Comment:
Could this assignment be done by the resource manager (in `Start`) rather than by every task type? That feels a little less error-prone to me, though I'm not sure.

Reply:
I'd rather put shared code in a different logic component (db) rather than the shared component (resource manager). And let the business logic (tasks) use the logic components respectively. I know there are shared code between tasks but I think it can be simplified by defining a task interface rather than putting shared code in the wrong logic component.

We used to put proxy, container management, resource management, task state, and etc, into the resource manager, which is so terrible because debugging individual logic components became impossible.

Reply:
I sort of see both sides here. A clean interface for tasks is nice and individual tasks being allowed to cleanly configure what they need will help us long term, but it is a little silly/error-prone to have the logic duplicated in every task. What if instead we had a `taskSpec.WithTaskToken` bool, as the interface, instead of tasks needing to know how to construct a task token? The only downside I see is the task spec needs a db to construct itself then.

Reply:
Either way, I think this is easy to change later if it becomes clear one is better than the other.